### PR TITLE
Optimized Profile stuff

### PR DESCRIPTION
**Pull request by rauchp**
- Made ProfileView fast.
- Made Edit ProfileView faster.
- Made Create Room Faster, added loading.
- Resolved Alex's nits for Profile.

### CODE REVIEW

File: civilapp/src/browse/SelectBannerScreen.js [hunk 1]
```diff
@@ -102,11 +101,28 @@ class SelectBannerScreen extends Component {
...
+        <Text style={styles.text}>Making Your Room...</Text>
+      </View>
+    </>
+  )
   render() {
     return (
       <RetroBackground>
-        {this.renderHeader()}
-        {this.renderBanners()}
+        {this.state.loadingRoomInit ? (
```
Comment (EphemeralX) [inline civilapp/src/browse/SelectBannerScreen.js]: nit: can we get into the habit of destructuring state/props?  i.e. `const { loading } = this.state`

File: civilapp/src/profile/ProfileEditScreen.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ class ProfileEditScreen extends React.Component {
   initScreen = async () => {
     const { navigation } = this.props
     const userId = navigation.getParam('userId')
-    const userInfo = await UserAPI.getUser(userId)
+    const userInfo = navigation.getParam('userInfo')
```
Comment (EphemeralX) [inline civilapp/src/profile/ProfileEditScreen.js]: I think this was there originally to ensure that the user info is always up-to-date, do you foresee this being a problem?

File: civilapp/src/profile/ProfileScreenNew.js [hunk 3]
```diff
@@ -181,7 +64,9 @@ class ProfileScreenNew extends React.Component {
       userId = userIdFromParams
       canEdit = false
     }
-    const userInfo = await UserAPI.getUser(userIdFromParams)
+
+    if (!canEdit) userInfo = await UserAPI.getUser(userId)
```
Comment (EphemeralX) [inline civilapp/src/profile/ProfileScreenNew.js]: nit: you guys may disagree but I find that even for 1 line if-statements, wrapping in `{}` is cleaner and more readable - I always do this for the code I touch/see.

File: civilapp/src/profile/ProfileEditScreen.js (hunk 2 above)
Comment (rauchp) [reply civilapp/src/profile/ProfileEditScreen.js]: My thinking - the only way to edit your profile is by getting past this screen. Which means you can't ever be passed stale data, (assuming we reload the userInfo that's passed in, I'm doing those hooks in my current PR now.)

File: civilapp/src/profile/ProfileScreenNew.js (hunk 3 above)
Comment (rauchp) [reply civilapp/src/profile/ProfileScreenNew.js]: nope I agree, I just didn't catch this.

File: civilapp/src/browse/CreateRoomScreen.js [hunk 4]
```diff
@@ -83,7 +80,7 @@ export default class CreateRoomScreen extends React.Component {
         leftButtonPress={this.backButtonPress}
         rightButtonPress={() => {
           navigation.navigate('SelectBanner', {
-            createRoom: this.initRoom.bind(this)
+            createRoom: (banner) => this.initRoom(banner)
```
Comment (carteryu) [inline civilapp/src/browse/CreateRoomScreen.js]: is this better for performance? j/w

Comment (rauchp) [reply civilapp/src/browse/CreateRoomScreen.js]: nope, I was testing this out and forgot 2 change. turns out that binding wasn't ever necessary, and with `createRoom: this.initRoom` we get the same results as before and can even parameter pass still.

**Pull request merged**